src/compiler/checker.ts

```diff
@@ -1985,6 +1986,100 @@ namespace ts {
             return false;
         }
 
+        function isTypeAccessible(type: Type, enclosingDeclaration: Node): boolean {
+            const meaning = SymbolFlags.Type;
+
+            return isTypeAccessibleWorker(type, /*inObjectLiteral*/ false, /*inTypeAlias*/true);
+
+            function isTypeAccessibleWorker(type: Type, inObjectLiteral: boolean, inTypeAlias: boolean): boolean {
+                if (!type) {
+                    return false;
+                }
+
+                if (inTypeAlias && type.aliasSymbol) {
+                    return isSymbolAccessible(type.aliasSymbol, enclosingDeclaration, SymbolFlags.Type, /*shouldComputeAliasesToMakeVisible*/false).accessibility === SymbolAccessibility.Accessible

```

Comment:
You use `SymbolFlags.Type` here, and `meaning` below (which you created as a const for SymbolFlags.Type on line 1990). Either get rid of the const, or use it consistently.